From ab1815b4dfa1ffb52f26fc2dc2e2394a86b7dd81 Mon Sep 17 00:00:00 2001
From: Martin Pitt <martin@piware.de>
Date: Sun, 14 Mar 2021 08:42:17 +0100
Subject: [PATCH 1/2] preload: Stop using deprecated input_event.time

The kernel deprecated this struct field to extend the range of 32 bit
time stamps:
https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/commit?id=152194fe9c3f

Part of issue #120
---
 src/libumockdev-preload.c |  2 +-
 tests/test-umockdev.c     | 20 ++++++++++----------
 2 files changed, 11 insertions(+), 11 deletions(-)

diff --git a/src/libumockdev-preload.c b/src/libumockdev-preload.c
index f1ae0e4..08ef2ba 100644
--- a/src/libumockdev-preload.c
+++ b/src/libumockdev-preload.c
@@ -1042,7 +1042,7 @@ script_record_op(char op, int fd, const void *buf, ssize_t size)
 	    const struct input_event *e = buf;
 	    while (size > 0) {
 		fprintf(srinfo->log, "E: %li.%06li %04"PRIX16" %04"PRIX16 " %"PRIi32"\n",
-			(long) e->time.tv_sec, (long) e->time.tv_usec, e->type, e->code, e->value);
+			(long) e->input_event_sec, (long) e->input_event_usec, e->type, e->code, e->value);
 		size -= sizeof(struct input_event);
 		e++;
 	    }
diff --git a/tests/test-umockdev.c b/tests/test-umockdev.c
index f664870..bffe550 100644
--- a/tests/test-umockdev.c
+++ b/tests/test-umockdev.c
@@ -1833,8 +1833,8 @@ t_testbed_replay_evemu_events(UMockdevTestbedFixture * fixture, gconstpointer da
   g_assert_cmpint(gettimeofday(&tv_end, NULL), ==, 0);
   assert_delta_t(&tv_begin, &tv_end, 10);
   tv_begin = tv_end;
-  g_assert_cmpint(ev.time.tv_sec, ==, 1234);
-  g_assert_cmpint(ev.time.tv_usec, ==, 500000);
+  g_assert_cmpint(ev.input_event_sec, ==, 1234);
+  g_assert_cmpint(ev.input_event_usec, ==, 500000);
   g_assert_cmpint(ev.type, ==, 0);
   g_assert_cmpint(ev.code, ==, 0);
   g_assert_cmpint(ev.value, ==, 0);
@@ -1842,8 +1842,8 @@ t_testbed_replay_evemu_events(UMockdevTestbedFixture * fixture, gconstpointer da
   /* read RELX event, should happen after 150 ms */
   g_assert_cmpint(read(fd, &ev, sizeof(ev)), ==, sizeof(ev));
   g_assert_cmpint(gettimeofday(&tv_end, NULL), ==, 0);
-  g_assert_cmpint(ev.time.tv_sec, ==, 1234);
-  g_assert_cmpint(ev.time.tv_usec, ==, 650000);
+  g_assert_cmpint(ev.input_event_sec, ==, 1234);
+  g_assert_cmpint(ev.input_event_usec, ==, 650000);
   g_assert_cmpint(ev.type, ==, 2);
   g_assert_cmpint(ev.code, ==, 0);
   g_assert_cmpint(ev.value, ==, 5);
@@ -1853,8 +1853,8 @@ t_testbed_replay_evemu_events(UMockdevTestbedFixture * fixture, gconstpointer da
   /* read RELY event, should happen immediately after */
   g_assert_cmpint(read(fd, &ev, sizeof(ev)), ==, sizeof(ev));
   g_assert_cmpint(gettimeofday(&tv_end, NULL), ==, 0);
-  g_assert_cmpint(ev.time.tv_sec, ==, 1234);
-  g_assert_cmpint(ev.time.tv_usec, ==, 649000);
+  g_assert_cmpint(ev.input_event_sec, ==, 1234);
+  g_assert_cmpint(ev.input_event_usec, ==, 649000);
   g_assert_cmpint(ev.type, ==, 2);
   g_assert_cmpint(ev.code, ==, 1);
   g_assert_cmpint(ev.value, ==, -11);
@@ -1864,8 +1864,8 @@ t_testbed_replay_evemu_events(UMockdevTestbedFixture * fixture, gconstpointer da
   /* read KEY_ZOOM event, should happen after 550 ms */
   g_assert_cmpint(read(fd, &ev, sizeof(ev)), ==, sizeof(ev));
   g_assert_cmpint(gettimeofday(&tv_end, NULL), ==, 0);
-  g_assert_cmpint(ev.time.tv_sec, ==, 1235);
-  g_assert_cmpint(ev.time.tv_usec, ==, 200000);
+  g_assert_cmpint(ev.input_event_sec, ==, 1235);
+  g_assert_cmpint(ev.input_event_usec, ==, 200000);
   g_assert_cmpint(ev.type, ==, 1);
   g_assert_cmpint(ev.code, ==, 0x174);
   g_assert_cmpint(ev.value, ==, 1);
@@ -1910,8 +1910,8 @@ t_testbed_replay_evemu_events_default_device(UMockdevTestbedFixture * fixture, g
 
   /* read SYN event; that should happen immediately */
   g_assert_cmpint(read(fd, &ev, sizeof(ev)), ==, sizeof(ev));
-  g_assert_cmpint(ev.time.tv_sec, ==, 1234);
-  g_assert_cmpint(ev.time.tv_usec, ==, 500000);
+  g_assert_cmpint(ev.input_event_sec, ==, 1234);
+  g_assert_cmpint(ev.input_event_usec, ==, 500000);
   g_assert_cmpint(ev.type, ==, 0);
   g_assert_cmpint(ev.code, ==, 0);
   g_assert_cmpint(ev.value, ==, 0);
-- 
2.30.2

